src/Dialect/ONNX/Rewrite.td

```diff
@@ -595,8 +599,9 @@ def RemoveIdentityReshapePattern1:  Pat<
   (ONNXReshapeOp $val, $shape, $az),
   // Remove the reshape.
   (replaceWithValue $val),
-  // Check that val has the specified shape.
-  [(HasSpecifiedConstantShape $val, $shape)]>;
+  // Check if shape has no value and that val has the specified shape.
+  [(IsNoneValue:$shape), (HasSpecifiedConstantShape $val, $shape)]
```

Comment:
`HasSpecifiedConstantShape` will never run if `shape` is not None (`IsNoneValue:$shape` returns false), so it's redundant, please remove it. Also `RemoveIdentityReshapePattern2` covered the situation of `HasSpecifiedConstantShape`.

Reply:
Please add a lit test for the case where `shape` is None.

Reply:
@tungld We can not use the value `None` in the reshape lit test... I get the following error:

```
within split at /Users/meganhampton/zDLC/onnx-mlir/test/mlir/onnx/onnx_canonicalization.mlir:203 offset :7:8: error: 'onnx.Reshape' op operand #1 must be tensor of 64-bit signless integer values, but got 'none'
  %1 = "onnx.Reshape"(%arg0, %cst) : (tensor<10x11x12x13xf32>, none) -> tensor<10x11x12x13xf32>
```

since onnx.Reshape does not have an optional value for reshape it automatically fails. Maybe you have another idea but I did not include the lit test for that reason :)

Reply:
Interesting! So by its definition onnx.Reshape does not allow None, so somehow onnx.Reshape with None existed in generated IR and passed its verification.

Do you know whether onnx.Reshape with None is in the original onnx model or generated by our compiler?